modules/command-engine/scaladsl/src/main/scala/surge/scaladsl/command/CommandModels.scala

```diff
@@ -2,97 +2,73 @@
 
 package surge.scaladsl.command
 
-import surge.core.command.AggregateCommandModelCoreTrait
-import surge.internal
-import surge.internal.domain.{ AsyncCommandHandler, CommandHandler }
-import surge.internal.persistence
-import surge.scaladsl.command.Types.CommandResult
+import surge.core.commondsl.SurgeProcessingModelCoreTrait
+import surge.internal.domain.{ SurgeContext, SurgeProcessingModel }
 import surge.scaladsl.common.Context
 
 import scala.concurrent.{ ExecutionContext, Future }
 import scala.util.Try
 
-trait AggregateCommandModel[Agg, Cmd, Evt] extends AggregateCommandModelCoreTrait[Agg, Cmd, Nothing, Evt] {
+trait AggregateCommandModel[Agg, Cmd, Evt] extends SurgeProcessingModelCoreTrait[Agg, Cmd, Evt] {
   def processCommand(aggregate: Option[Agg], command: Cmd): Try[Seq[Evt]]
   def handleEvent(aggregate: Option[Agg], event: Evt): Option[Agg]
 
-  final def toCore: CommandHandler[Agg, Cmd, Nothing, Evt] =
-    new CommandHandler[Agg, Cmd, Nothing, Evt] {
-      override def processCommand(ctx: persistence.Context, state: Option[Agg], cmd: Cmd): Future[CommandResult] =
-        Future.fromTry(AggregateCommandModel.this.processCommand(state, cmd).map(v => Right(v)))
-      override def apply(ctx: persistence.Context, state: Option[Agg], event: Evt): Option[Agg] = handleEvent(state, event)
+  final def toCore: SurgeProcessingModel[Agg, Cmd, Evt] = {
+    new SurgeProcessingModel[Agg, Cmd, Evt] {
+      override def handle(ctx: SurgeContext[Agg, Evt], state: Option[Agg], msg: Cmd)(implicit ec: ExecutionContext): Future[SurgeContext[Agg, Evt]] = {
+        Future.fromTry(processCommand(state, msg)).map { events =>
+          val newState = events.foldLeft(state)((s: Option[Agg], e: Evt) => handleEvent(s, e))
+          ctx.persistEvents(events).updateState(newState).reply(state => state)
+        }
+      }
+
+      // FIXME applyAsync everywhere needs a Future.fromTry to handle exceptions thrown by the domain
```

Comment:
Need to double check this. This was being (sorta magically) handled by one of the layers before but must be handled at this layer everywhere now